### PR TITLE
Ci/matrix

### PR DESCRIPTION
**Pull request by BotellaA**
Add build on ubuntu using g++ or clang++, on macOS using AppleClang or Xcode

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -37,26 +49,45 @@ jobs:
...
+      uses: pullreminders/slack-action@master
+      with:
+        args: '{\"channel\":\"CLJ1PS7UN\",\"text\":\"Compilation or Testing failed for OpenGeode on Ubuntu\"}'        
       
   build-mac:
     runs-on: macOS-10.14
     needs: format
+    strategy:
+      matrix:
+        cmake_args: ['-DCMAKE_BUILD_TYPE=Debug', '-G "Xcode"']
```
Comment (benjaminchauvin) [inline .github/workflows/CI.yml]: Not sure that you need "" around Xcode.

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -37,26 +49,45 @@ jobs:
...
     steps:
     - uses: actions/checkout@v1
     - name: Compile & Test
       run: |
         mkdir -p build
         cd build
-        cmake -DCMAKE_BUILD_TYPE=Debug ..
-        cmake --build . -- -j2
+        cmake ${{ matrix.cmake_args }} ..
+        cmake --build . --config Debug
```
Comment (benjaminchauvin) [inline .github/workflows/CI.yml]: You do not need Debug when you use clang without Xcode. I presume that it does not hurt but is it possible to remove it when it is clang without Xcode (to be cleaner)?

File: .github/workflows/CI.yml [hunk 3]
```diff
@@ -37,26 +49,45 @@ jobs:
...
         cd opengeode
-        ctest --output-on-failure
-      
+        ctest -C Debug --output-on-failure
```
Comment (benjaminchauvin) [inline .github/workflows/CI.yml]: Same comment about the "debug".

**Comment by benjaminchauvin** [comment]
There is no check in Release? It happens that it passes in debug and not in release.

File: .github/workflows/CI.yml (hunk 1 above)
Comment (BotellaA) [reply .github/workflows/CI.yml]: Yes I do. I tried without them first and it did not work

File: .github/workflows/CI.yml (hunk 2 above)
Comment (BotellaA) [reply .github/workflows/CI.yml]: I know but I did not find à clean way to do it yet... 😕

**Comment by BotellaA** [comment]
No checks in release because it takes time and harder to implement. Not sure if it is really necessary. @panquez what do you think ?

**Comment by benjaminchauvin** [comment]
> No checks in release because it takes time and harder to implement. Not sure if it is really necessary. @panquez what do you think ?

I do think that it is really important for a (surely more) very important reason: users will use the release version, not the debug one. Ensuring that the release works should be the priority. The debug version is just a tool for the developers. You should have both. This is just a personal opinion and I know that doing both complexifies everything (but for a good goal).

**Pull request merged**